packages/turbo/src/stradaBridge.ts

```diff
@@ -0,0 +1,223 @@
+    }
+
+    unregister(component) {
+      const index = this.supportedComponents.indexOf(component);
+      if (index != -1) {
+        this.supportedComponents.splice(index, 1);
+        this.notifyBridgeOfSupportedComponentsUpdate();
+      }
+    }
+
+    registerAdapter() {
+      this.adapterIsRegistered = true;
+
+      if (this.isStradaAvailable) {
+        this.webBridge.setAdapter(this);
+      } else {
+        document.addEventListener('web-bridge:ready', () =>
+          this.webBridge.setAdapter(this)
+        );
+      }
+    }
+
+    notifyBridgeOfSupportedComponentsUpdate() {
+      if (this.isStradaAvailable) {
+        this.webBridge.adapterDidUpdateSupportedComponents();
+      }
+    }
+
+    supportsComponent(component) {
+      return this.supportedComponents.includes(component);
+    }
+
+    // Reply to web with message
+    replyWith(message) {
+      if (this.isStradaAvailable) {
+        this.webBridge.receive(JSON.parse(message));
+      }
+    }
+
+    // Receive from web
+    receive(message) {
+      this.postMessage(message);
+    }
+
+    get platform() {
+      return '${Platform.OS}';
+    }
+
+    // Native handler
+
+    postMessage(message) {
+      ${
+        Platform.OS === 'android'
+          ? 'StradaNative.postMessage(JSON.stringify(message))'
+          : 'webkit.messageHandlers.stradaNative.postMessage(message)'
+      }
+    }
+
+    // Web global
+
+    get isStradaAvailable() {
+      return window.Strada;
+    }
+
+    get webBridge() {
+      return window.Strada.web;
+    }
+  }
+
+  if (
+    document.readyState === 'interactive' ||
+    document.readyState === 'complete'
+  ) {
+    initializeBridge();
+  } else {
+    document.addEventListener('DOMContentLoaded', () => {
+      initializeBridge();
+    });
+  }
+
+  function initializeBridge() {
+    window.nativeBridge = new NativeBridge();
+  }
+})();
+`;
+
+const useStradaBridge = (
+  url: string,
+  sessionHandle: string,
+  stradaComponents?: StradaComponent[]
+) => {
+  const initializeStradaBridge = useCallback(async () => {
+    const isStradaUndefined = await RNVisitableViewModule.injectJavaScript(
+      sessionHandle,
+      `window.nativeBridge == undefined`
+    );
+    if (isStradaUndefined) {
+      await RNVisitableViewModule.injectJavaScript(
+        sessionHandle,
+        stradaBridgeScript
+      );
+      if (!stradaComponents) {
+        return;
+      }
+      Promise.all(
+        stradaComponents.map(({ name }) =>
+          RNVisitableViewModule.injectJavaScript(
+            sessionHandle,
+            `window.nativeBridge.register('${name}')`
+          )
+        )
+      );
+    }
+  }, [sessionHandle, stradaComponents]);
+
+  const stradaUserAgent = useMemo(
+    () =>
+      stradaComponents
+        ? ` bridge-components: [${stradaComponents.reduce(
+            (acc, { name }) => `${acc} ${name}`,
+            ''
+          )} ]`
+        : '',
+    [stradaComponents]
+  );
```

Comment:
This is really hard to read and reason about.

What about using `map` and `join` for the concatenation? Could also do an early return if no `stradaComponents` is nil.

Reply:
```js
const stradaUserAgent = useMemo(() => {
  if (!stradaComponents) return;

  const componentNames = stradaComponents.map(({ name }) => name);
  return `bridge-components: [${componentNames.join(" ")}]`;
}, [stradaComponents]);